### PR TITLE
DRILL-6389: Generate Javadoc for all classes and fix some warnings.

### PR DESCRIPTION
**Pull request by ilooner**
# Fixes For Javadoc Generation

This patch accomplishes the following:

- Javadoc is generated for all classes (except for a few excluded classes).
- Javadoc warnings about unknown classes and packages are fixed.
- Some (but not all) easy javadoc warnings are fixed.
- Documentation was added on how to add and view javadocs

## Implementation Details

The javadoc plugin can only compile and link all the javadocs if `mvn javadoc:aggregate` is called in the project root directory. The javadoc plugin is also unaware of the test dependencies, so they must be explicitly added to the plugin in order for javadoc to be able to resolve test classes.

Also since test dependencies must be explicitly added to the javadoc plugin, we must also create test jars for drill-common and drill-java-exec which contain common test classes used by other modules, and add these test-jars to the javadoc plugin dependency list.

Some classes were excluded from javadoc generation due to weird package linking issues I could not figure out resulting in javadoc warnings.

There was also a bug with the maven-license-plugin. The plugin definition was included in the plugins section not the pluginManagement section of the pom. This caused a strange issue where an extra license header was being added in storage-kudu/src/main/codegen/templates/KuduRecordWriter.java . This caused the javadoc tool to throw an error.

### CODE REVIEW

**Comment by ilooner** [comment]
@arina-ielchiieva

File: pom.xml [hunk 1]
```diff
@@ -202,6 +208,138 @@
...
+            <!-- Don't include codegen -->
+            <exclude>**/src/main/codegen/**/*.java</exclude>
+            <!-- Don't include target -->
+            <exclude>**/target/**/*.java</exclude>
+            <!--
+              Note: There is a bug with javadoc that causes javadoc to be unable to find the org.apache.parquet.hadoop.CodecFactory class in the parquet-hadoop dependency. This
+              is because we have declared our own org.apache.parquet.hadoop package in exec/java-exec. As a result we have to exclude some parquet classes, which generate
+              javadoc warnings because of this.
+            -->
+            <!-- Exclude files with strange package linking issues -->
```
[the remark this answers is not in the document]
Comment (ilooner) [reply pom.xml]: @arina-ielchiieva I agree this is not ideal. I spent some more time trying to resolve the warnings introduced by these files and couldn't find a solution. I've removed the excludes so now we will still generate javadoc for these files. The only negative consequence to this is that there will be a few more javadoc warnings when running `mvn javadoc:aggregate`.

Would you be okay with removing the excludes and resolving the extra javadoc warnings in a separate Jira?

**Comment by ilooner** [comment]
@arina-ielchiieva Could you take another look at this?

File: pom.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (arina-ielchiieva) [reply pom.xml]: Yes, I think that should be fine. Thanks!

**Comment by arina-ielchiieva** [comment]
+1, please add comment with Jira number added for the resolving warnings (create Jira if not yet created).

**Pull request closed without merging**